testutil/server.go

```diff
@@ -195,6 +196,7 @@ func NewTestServer(t testing.T, cb ServerConfigCallback) *TestServer {
 	if nomadConfig.Stderr != nil {
 		stderr = nomadConfig.Stderr
 	}
+	t.Logf("CONFIG JSON: %s", string(configContent))
```

Comment:
This now matches the equivalent logging we have for test Consul servers, and feels pretty nice when I've been working on https://github.com/hashicorp/nomad/pull/18799